### PR TITLE
[MRG] fix circle, add subject attr in make_stc_from_dipoles and put src as 1st param

### PR DESCRIPTION
**Pull request by agramfort**
attempt to fix circle + raise discussion on API.

### CODE REVIEW

File: examples/inverse/plot_mixed_norm_inverse.py [hunk 1]
```diff
@@ -82,11 +82,11 @@
...
-# # Plot dipole locations of all dipoles with MRI slices
-# for dip in dipoles:
-#     plot_dipole_locations(dip, forward['mri_head_t'], 'sample',
-#                           subjects_dir=subjects_dir, mode='orthoview',
-#                           idx='amplitude')
+# Plot dipole locations of all dipoles with MRI slices
+for dip in dipoles:
+    plot_dipole_locations(dip, forward['mri_head_t'], 'sample',
+                          subjects_dir=subjects_dir, mode='orthoview',
+                          idx='amplitude')
```
Comment (agramfort) [inline examples/inverse/plot_mixed_norm_inverse.py]: I don't know why this was commented out

File: mne/morph.py [hunk 2]
```diff
@@ -416,6 +416,8 @@ def save(self, fname, overwrite=False, verbose=None):
 def _check_subject_from(subject_from, src):
     if isinstance(src, string_types):
         subject_check = src
+    elif src is None:  # assume it's correct although dangerous but unlikely
+        subject_check = subject_from
```
Comment (agramfort) [inline mne/morph.py]: if you have an stc you give as src in compute_source_morph and this stc has no subject then you now assume it's coming from subject_from and don't break

File: tutorials/plot_stats_cluster_spatio_temporal_2samp.py [hunk 3]
```diff
@@ -40,8 +40,11 @@
 # Read the source space we are morphing to
 src = mne.read_source_spaces(src_fname)
 fsave_vertices = [s['vertno'] for s in src]
-stc = mne.morph_data('sample', 'fsaverage', stc, grade=fsave_vertices,
```
Comment (agramfort) [inline tutorials/plot_stats_cluster_spatio_temporal_2samp.py]: this is deprecated

File: doc/manual/cookbook.rst [hunk 4]
```diff
@@ -429,6 +429,9 @@ Group analyses
 Group analysis is facilitated by morphing source estimates, which can be
 done *e.g.*, to ``subject='fsaverage'`` as::
 
-    >>> stc_fsaverage = stc.morph('fsaverage')  # doctest: +SKIP
+    >>> morph = mne.compute_source_morph(subject_from='sample',
+                                         subject_to='fsaverage',
+                                         src=stc)  # doctest: +SKIP
+    >>> stc_fsaverage = morph.apply(stc)  # doctest: +SKIP
```
Comment (agramfort) [inline doc/manual/cookbook.rst]: @larsoner this illustrates a current concern with the new API. You **must** provide a src parameter. 2 issues:

  - It cannot in fact be None although it's None in function signature.

  - you need to pass stc twice...

cc @TommyClausner

Comment (larsoner) [reply doc/manual/cookbook.rst]: Having to pass STC twice is fine by me. But we should not allow it to be none. One option is to actually make it the first argument

Comment (agramfort) [reply doc/manual/cookbook.rst]: +1 to make it the first argument with no default value. Can you push this to this branch if you have time?

Comment (larsoner) [reply doc/manual/cookbook.rst]: I can do it early next week

**Comment by larsoner** [comment]
Thanks @agramfort

**Pull request merged**